content/en/debugging.md

```diff
@@ -0,0 +1,14 @@
+---
+title: Debugging
+status: Feedback Appreciated
+category: concept
+---
+
+## What it is
+Debugging is the process or activity of finding and resolving bugs (or errors) from computer programs, software, or systems to get the desired result. A bug is a defect or a problem leading to incorrect or unexpected results.
+
+## Problem it addresses
+Software development is a complex activity that makes it nearly impossible to write code without introducing bugs. Those bugs lead to code that will likely not function as desired (undefined behavior) when executed. Depending on how critical an application is, bugs can have a significant negative impact — financially or even on human lives. Usually, application code has to go through different stages or environments where it gets tested. The more critical an application is, the more accurate the testing has to be. 
+
+## How it helps
+When bugs appear, engineers have to debug (e.g., finding and fixing) the app to decrease undesired behavior for production systems. Debugging is no easy task as engineers have to track down the source of the undesired behavior. It requires knowledge about the code itself and the execution context at runtime. This is where different debugging techniques and tools come in handy. Analysis of logs, traces, and metrics, for instance, are used for debugging directly in production. In contrast, developers can use interactive debugging to step through the code at runtime while analyzing the related execution context. Once they have identified the source of the failure, they correct the code and create a bug fix or patch.
```

Comment:
could be my fault, as I made a little edit because some suggestions were merged resulting in a strange sentence. I 
Agree with Catherine's suggestion :)

Reply:
fixed :D thx for the suggestion